### ISSUE TITLE
FR: Magazine Filter Blocks

### ISSUE THREAD

**Issue by ZoeBeykirch** [hlxsites/vg-macktrucks-com#808]
### Context/Description

As a user visiting the Bulldog Magazine website, I want to filter based on categories, topics, and truck models so that I can easily find relevant articles that interest me. 

### Acceptance Criteria

**User AC:**
- [ ] Users can view 3 dedicated filter dropdown list containing the following options: All Categories, All Topics, All Trucks.
- [ ] On desktop/tablets, the category filter bars are presented on the same row side by side.
- [ ] On mobile devices, the category filter bars stack on top of each other. 
- [ ] All category filters are the same height and size. 
- [ ] When the user selects the arrow on the category bar, a dropdown list will appear with the following options:

  **1. All Categories** 
       - Customer Profiles 
       - In the Know
       - Product Updates 
       - Technology at Work
       - Features 
       - In the News 
       - Viewpoint
       - Calendar 

    **2. All Topics** 
       - Topic options are TBD @WendyKruger 
    
  **3. All Trucks** 
       - Anthem
       - Granite
       - LR
       - LR Electric
       - MD 
       - MD Electric
       - Pinnacle
       - TerraPro

- [ ] Users should be able to select multiple options for each dropdown. For example, a user can select multiple truck models and topics at the same time. 

**Non-functional AC:**
- [ ] Upon selecting filters, the article content on the page should dynamically update to display articles that match all selected criteria.
- [ ] The filtering should work efficiently without page reloads.
- [ ] The users shall see the categories they have selected in the bar when the content populates. 
- [ ] If the user selects multiple category options, there will be a character limit of around 20 and ellipses after the limit. So for example if the user selects 3 trucks the filter shall display: Anthem, Pinnacle, Gra...
- [ ] The filter drop down should have a scroll bar if the user's screen is not large enough to show the entire list. 

### Out of scope

Search functionality for magazine will be handled in a [separate ticket](https://github.com/hlxsites/vg-macktrucks-com/issues/766)
Sort by feature will be handled in a separate ticket 

### Note

Should the search bar styling (not functionality) be taken up in this ticket or another dedicated ticket?

### Technical details

We want to avoid what is currently happening on production for the All Truck models example where if an article is tagged with multiple product tags that it creates a separate tag line as seen here: https://www.macktrucks.com/magazine/explore-articles

![Image](https://github.com/user-attachments/assets/a27bbce5-a202-48de-a959-3ad118f83a17)

### Attachment / link to Figma

Desktop: https://www.figma.com/design/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?node-id=15841-4009&m=dev

![Image](https://github.com/user-attachments/assets/3c41d120-c0c3-4a96-8173-31423e19386f)

Mobile: https://www.figma.com/design/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?node-id=15794-4007&m=dev

![Image](https://github.com/user-attachments/assets/db1061eb-af2d-4667-b2a5-1a358108ed53)

![Image](https://github.com/user-attachments/assets/fb0e06fe-ef85-4252-af4f-034ae0c69d70)

**Comment by mollyaljenkins** [comment]
This ticket is dependent on the BE search efforts

**Comment by mollyaljenkins** [comment]
To be updated based on #602

**Comment by BeckyMedlin** [comment]
@WendyKruger Please see design question from Burundi in Slack: https://adobe-dx-support.slack.com/archives/C05JATBN087/p1726585782343469

**Comment by bfvolvo** [comment]
Adding in the links

> @WendyKruger Please see design question from Burundi in Slack: https://adobe-dx-support.slack.com/archives/C05JATBN087/p1726585782343469

Appears we received approval from @WendyKruger here
https://adobe-dx-support.slack.com/archives/C05JATBN087/p1726755874321499?thread_ts=1726596820.654929&cid=C05JATBN087

Adding in a direct link to the Figma file, please contact me if any questions or need access
https://www.figma.com/proto/UDwzhFngA1NNhObvJLLsTr/Mack-Truck-Design-Recommendations?page-id=0%3A1&node-id=43-1038&node-type=frame&viewport=691%2C597%2C0.18&t=bbI3xGh5oNx7zb0P-1&scaling=min-zoom&content-scaling=fixed

**Comment by santi-homps** [comment]
@bfvolvo I requested access to the figma link in the comment above, could you provide it please?

**Comment by bfvolvo** [comment]
> @bfvolvo I requested access to the figma link in the comment above, could you provide it please?

@santi-homps I have provided access here is a Figma Dev Mode link which may be better for you - https://www.figma.com/design/UDwzhFngA1NNhObvJLLsTr/Mack-Truck-Design-Recommendations?node-id=43-1038&m=dev

**Comment by ZoeBeykirch** [comment]
@WendyKruger we are still waiting on what drop down options will be available for the "All Topics" filter.

**Comment by ZoeBeykirch** [comment]
Moving back to refinement to re-evaluate our estimates, taking into account the new requirements that allow content editors to edit, add, and remove filter tags. We'll also address the new question related to tag management.

**Comment by mollyaljenkins** [comment]
Moved out of the sprint and blocked pending updates from @bfvolvo and @WendyKruger on design
